homeassistant/components/climate/hive.py

```diff
@@ -0,0 +1,204 @@
+                        self.session.data.minmax[nodeid]['TodayMin'] = 1000
+                        self.session.data.minmax[nodeid]['TodayMax'] = -1000
+                        self.session.data.minmax[nodeid]['TodayDate'] = \
+                            datetime.date(datetime.now())
+
+                    if (curtempret < self.session.data.minmax[nodeid]
+                            ['TodayMin']):
+                        self.session.data.minmax[nodeid]['TodayMin'] = \
+                            curtempret
+
+                    if (curtempret > self.session.data.minmax[nodeid]
+                            ['TodayMax']):
+                        self.session.data.minmax[nodeid]['TodayMax'] = \
+                            curtempret
+
+                    if (curtempret < self.session.data.minmax[nodeid]
+                            ['RestartMin']):
+                        self.session.data.minmax[nodeid]['RestartMin'] = \
+                            curtempret
+
+                    if (curtempret >
+                            self.session.data.minmax[nodeid]
+                            ['RestartMax']):
+                        self.session.data.minmax[nodeid]['RestartMax'] = \
+                            curtempret
+                else:
+                    current_node_max_min_data = {}
+                    current_node_max_min_data['TodayMin'] = curtempret
+                    current_node_max_min_data['TodayMax'] = curtempret
+                    current_node_max_min_data['TodayDate'] = \
+                        datetime.date(datetime.now())
+                    current_node_max_min_data['RestartMin'] = curtempret
+                    current_node_max_min_data['RestartMax'] = curtempret
+                    self.session.data.minmax[nodeid] = \
+                        current_node_max_min_data
+            else:
+                curtempret = 0
+            return curtempret
+        elif self.device_type == "HotWater":
+            return None
+
+    @property
+    def target_temperature(self):
+        """Return the target temperature."""
+        set_result = None
+
+        if self.device_type == "Heating":
+            set_result = self.session.heating.get_target_temperature(
+                self.node_id)
+        elif self.device_type == "HotWater":
+            set_result = None
+
+        return set_result
+
+    @property
+    def min_temp(self):
+        """Return minimum temperature."""
+        if self.device_type == "Heating":
+            return self.session.heating.min_temperature(self.node_id)
+        elif self.device_type == "HotWater":
+            return None
+
+    @property
+    def max_temp(self):
+        """Return the maximum temperature."""
+        if self.device_type == "Heating":
+            return self.session.heating.max_temperature(self.node_id)
+        elif self.device_type == "HotWater":
+            return None
+
+    @property
+    def operation_list(self):

```

Comment:
You should translate the values returned by the library into the `STATE_*` constants from `climate/__init__.py`. And please use "auto" instead of "schedule".

Reply:
Sorry, I don't understand. Please could you give an example of an existing climate component which could help show me what you mean?

The value "schedule" is used as this is what is used in the official Hive application and website.

Reply:
Great, thanks, will take a detailed look.

So (after a quick look) I need / can to use the following mapping?
heating : auto = schedule, heat = manual, off = off
hotwater : auto = schedule, on = on, off = off

from the ```eq3btsmart``` platform it uses ```STATE_MANUAL``` but this is not imported from ```homeassistant.components.climate```, so I shouldn't use this?

Reply:
I have updated as above.
Is this ok now?

Reply:
Looks good. If you want to use `STATE_MANUAL`, you should probably define it in `climate/hive.py`. In some future refactoring, it will be merged with the `STATE_MANUAL` from eq3btsmart (and others).

Reply:
great, thanks.
I will have a think on the ``STATE_MANUAL``` vs ```STATE_HEAT```